### DISCUSSION TITLE
what's the usage of kafka-logger's plugin argument `meta_format`?

### DISCUSSION THREAD

**Discussion by ryan4yin** [apache/apisix#7840] [Q&A] [answered]
The plugin's document does not clearly explain the function of `meta_format`.

I understand that the format of the log is controlled by `log_format` in plugin's metadata,
so does setting `meta_format` to `default` or `origin` have any effect on access logs sent by kafka-logger?

**Comment by tokers** [comment, marked as answer]
Yes.

**Reply by ryan4yin** [reply]
suppose I set `log_format` by the following command:

```shell
curl http://127.0.0.1:9080/apisix/admin/plugin_metadata/kafka-logger -H 'X-API-KEY: edd1c9f034335f136f87ad84b625c8f1' -X PUT -d '
{
    "log_format": {
        "host": "$host",
        "@timestamp": "$time_iso8601",
        "client_ip": "$remote_addr"
    }
}'
```

what's effect will it take when I change `meta_format` from `origin` to `default`?

**Reply by tokers** [reply]
The priority is lower than the `meta_format` in plugin config.

**Reply by ryan4yin** [reply]
so you mean the function of `log_format` and `meta_format` is the same, and `meta_format` has higher priority than `log_format`?

**Reply by tokers** [reply]
Right, they are both used for the log format.
